src/btree/row_key.c

```diff
@@ -337,28 +373,26 @@ __wt_row_leaf_key_work(
              * find a key without a prefix.
              */
             if (slot_offset == 0)
-                last_prefix = unpack->prefix;
-            if (slot_offset == 0 || last_prefix > unpack->prefix) {
+                last_prefix = key_prefix;
+            if (slot_offset == 0 || last_prefix > key_prefix) {
                 jump_rip = rip;
                 jump_slot_offset = slot_offset;
-                last_prefix = unpack->prefix;
+                last_prefix = key_prefix;
             }
         }
         if (direction == FORWARD) {
-            p = unpack->data;
-            size = unpack->size;
-
             /*
              * Grow the buffer as necessary as well as ensure data has been copied into local buffer
              * space, then append the suffix to the prefix already in the buffer.
              *
              * Don't grow the buffer unnecessarily or copy data we don't need, truncate the item's
-             * data length to the prefix bytes.
+             * CURRENT data length to the prefix bytes before growing the buffer.
```

Comment:
Do you intend to leave `CURRENT` in all caps?

Reply:
Yes, all caps was deliberate.

People (including me!!), keep removing line #1016 because line #1019 sets it again. And, uh, that's wrong and I'm tired of debugging it. :)
```
   1016             tmpkey->size = key_prefix;
   1017             WT_ERR(__wt_buf_grow(session, tmpkey, key_prefix + key_size));
   1018             memcpy((uint8_t *)tmpkey->mem + key_prefix, key_data, key_size);
   1019             tmpkey->size = key_prefix + key_size;
```